personalcapital/personalcapital.js

```diff
@@ -207,6 +208,38 @@ class PersonalCapital {
       accounts, startDate, endDate, intervalType, types));
     return res;
   }
+
+  // accountName must match name on account exactly
+  async updateBalance(accountName /*String*/, newBalance /*Number*/) {
+    const account = await this.getAccountByName(accountName);
+    const updateData = {
+      ...account,
+      isTransferPending: false,
+      isTransferEligible: true,
+      ownershipType: null,
+      employerMatchLimitType: "dollar",
+      requestSource: "USER",
+      balance: newBalance,
+      currentBalance: newBalance,
+      availableBalance: newBalance
+    };
+    const res = await this.endpoint("updatebalance", updateData);
+    return res;
+  }
+
+  async getAccountByName(accountName /*String*/) {
+    const accounts = await this.getAccounts();
+
+    const selectedAccount = accounts.filter(account => {
+      return account.name === accountName;
+    });
+    if (!selectedAccount || selectedAccount.length === 0) {
+      throw `Account ${accountName} not found!`;
+    }
+
+    return selectedAccount[0];
+  }
```

Comment:
Make this more concise with something like this:
```js
async getAccountByName(accountName /*String*/) {
    const accounts = await this.getAccounts();
    const selectedAccount = accounts.find(account => account.name === accountName);
    if (!selectedAccount) {
      throw new AccountNotFoundError(`Account ${accountName} not found!`);
    }
    return selectedAccount;
  }
```

Reply:
Forgot but AccountNotFoundError should be defined on line 35 as `class AccountNotFoundError extends Error {}` and exported in the module.exports.